test/e2e/e2e_local_queue_defaulting_test.go

```diff
@@ -31,91 +32,55 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+var (
+	ns                   *corev1.Namespace
+	initialKueueInstance *kueueoperatorv1.Kueue
+	kueueClient          *upstreamkueueclient.Clientset
+	builder              *testutils.TestResourceBuilder
+)
+
 var _ = Describe("LocalQueueDefaulting", Ordered, func() {
+
 	BeforeAll(func() {
 		Expect(deployOperand()).To(Succeed(), "operand deployment should not fail")
 	})
 	AfterAll(func() {
 		testutils.CleanUpKueuInstance(context.TODO(), clients.KueueClient, "cluster")
 	})
 
-	When("labelPolicy=None and LocalQueue default", func() {
-		var (
-			ns                   *corev1.Namespace
-			initialKueueInstance *kueueoperatorv1.Kueue
-			kueueClient          *upstreamkueueclient.Clientset
-			builder              *testutils.TestResourceBuilder
-		)
+	When("labelPolicy=None and LocalQueue default in a managed namespace", func() {
 
 		BeforeAll(func() {
-			ctx := context.TODO()
-			kueueInstance, err := clients.KueueClient.KueueV1().Kueues().Get(ctx, "cluster", metav1.GetOptions{})
-			Expect(err).ToNot(HaveOccurred(), "Failed to fetch Kueue instance")
-			initialKueueInstance = kueueInstance.DeepCopy()
-			kueueInstance.Spec.Config.WorkloadManagement.LabelPolicy = kueueoperatorv1.LabelPolicyNone
-			applyKueueConfig(ctx, kueueInstance.Spec.Config, kubeClient)
-			kueueClient = clients.UpstreamKueueClient
-			Expect(testutils.CreateClusterQueue(kueueClient)).To(Succeed(), "Failed to create cluster queue")
-			Expect(testutils.CreateResourceFlavor(kueueClient)).To(Succeed(), "Failed to create resource flavor")
+			createClusterQueueAndResourceFlavor(context.TODO(), string(kueueoperatorv1.LabelPolicyNone))
```

Comment:
The BeforeAll, When, If, etc ginkgo functions support passing a context in the function. ie:

```
It("should label and admit Pod and Job", func(ctx context.Context) {
```

This will solve creating context.TODO() or context.Background() within the tests, and allow the test framework to set an overall timeout.